gk/main.c

```diff
@@ -543,7 +473,12 @@ gk_proc(void *arg)
 			struct flow_entry *fe;
 			struct rte_mbuf *pkt = rx_bufs[i];
 			
-			extract_packet_info(pkt, &packet);
+			ret = extract_packet_info(pkt, &packet);
+			if (ret < 0) {
+				/* Drop non-IP packets. */
+				rte_pktmbuf_free(pkt);
```

Comment:
Shouldn't you use `drop_packet()` here instead of `rte_pktmbuf_free()`? It's not a big deal, but given that you've defined `drop_packet()` before, the code should be consistent.